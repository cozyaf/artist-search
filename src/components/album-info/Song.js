import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Song({ song }) {
  const { title, artistName } = song;
  return (
    <Link to={`/lyrics/${artistName}/${title}`} style={{ textDecoration: 'none' }}>
      <section>
        <p>{title}</p>
      </section>
    </Link>
  );
}

Song.propTypes = {
  song: PropTypes.object.isRequired
};

export default Song;
