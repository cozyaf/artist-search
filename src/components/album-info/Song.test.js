import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song';

describe('Song component', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song song={{ title:'ere', artistName:'456' }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
