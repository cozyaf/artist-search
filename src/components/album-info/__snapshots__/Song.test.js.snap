// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component renders Song 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/lyrics/456/ere"
>
  <section>
    <p>
      ere
    </p>
  </section>
</Link>
`;
